pkg/build/generator/generator.go

```diff
@@ -293,6 +294,8 @@ func (g *BuildGenerator) generateBuildFromConfig(ctx kapi.Context, bc *buildapi.
 	}
 
 	build.Name = getNextBuildName(bc)
+	build.Annotations = make(map[string]string)
+	build.Annotations[buildapi.BuildNumberAnnotation] = strconv.Itoa(bc.Status.LastVersion)
```

Comment:
Right now clone builds work as before. No annotation, the long name appears in the UI.

I know we have an issue for the weird name, #3368.